src/rust-crypto/rust-crypto.ts

```diff
@@ -206,8 +206,22 @@ export class RustCrypto implements CryptoBackend {
     }
 
     public async exportRoomKeys(): Promise<IMegolmSessionData[]> {
-        // TODO
-        return [];
+        const raw = await this.olmMachine.exportRoomKeys(() => true);
+        return JSON.parse(raw);
+    }
+
+    public async importRoomKeys(keys: IMegolmSessionData[], opts?: ImportRoomKeysOpts): Promise<void> {
+        const jsonKeys = JSON.stringify(keys);
+        await this.olmMachine.importRoomKeys(jsonKeys, (progress: BigInt, total: BigInt) => {
+            const importOpt: ImportOpts = {
+                total: Number(total),
+                successes: Number(progress),
+                stage: "load_keys",
+                failures: 0,
+            };
```

Comment:
I'm assuming there was more than one `stage` in the legacy implementation - do we not need to replicate them? And do something with `failures`?

Reply:
In legacy it's the same, the stage is `load_keys` in the crypto module.
But the matrix client has a `restoreKeyBackup` method with a free form `opt` that has a similar `progressCallback` property and in this case when downloading from server the stage is `fetch`.

This would need to be refactored when implementing server backup I think.